### ISSUE TITLE
Connot RDP to CentOS 8 with valid AD account that works for SSH

### ISSUE THREAD

**Issue by TDS1969** [neutrinolabs/xrdp#1750]
I have installed xrdp on a CentOS 8 server with eh goal of RDPing from my companies MAC and PC laptops for support. The server is joined to our companies Active Directory domain for user accounts. These domain accounts work fine for ssh. I have installed XRDP and am able to initiate the connection, but it is failing on the actual login.  

This is what I am seeing in the xrdp.log file:

```
[20201211-12:56:11] [INFO ] connected client computer name: c00633
[20201211-12:56:11] [INFO ] adding channel item name rdpdr chan_id 1004 flags 0x80800000
[20201211-12:56:11] [INFO ] adding channel item name rdpsnd chan_id 1005 flags 0xc0000000
[20201211-12:56:11] [INFO ] adding channel item name cliprdr chan_id 1006 flags 0xc0a00000
[20201211-12:56:11] [INFO ] adding channel item name drdynvc chan_id 1007 flags 0xc0800000
[20201211-12:56:11] [INFO ] TLS connection established from 10.235.33.200 port 58836: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384
[20201211-12:56:12] [DEBUG] xrdp_00001fe4_wm_login_mode_event_00000001
[20201211-12:56:12] [INFO ] Loading keymap file /etc/xrdp/km-00000409.ini
[20201211-12:56:12] [WARN ] local keymap file for 0x00000409 found and doesn't match built in keymap, using local keymap file
[20201211-12:56:12] [DEBUG] xrdp_wm_log_msg: connecting to sesman ip 127.0.0.1 port 3350
[20201211-12:56:12] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20201211-12:56:12] [DEBUG] xrdp_wm_log_msg: sending login info to session manager, please wait...
[20201211-12:56:12] [DEBUG] return value from xrdp_mm_connect 0
[20201211-12:56:12] [INFO ] xrdp_wm_log_msg: login failed for display 0
[20201211-12:56:12] [DEBUG] xrdp_mm_module_cleanup
[20201211-12:56:12] [DEBUG] Closed socket 18 (AF_INET 127.0.0.1:44426)
```
I have seen on the internet where people have suggested that the "login failed for display 0" message is related to a bad login or password.  As I mentioned,t these work fine for SSH.  So I am at a loss.

I am not sure what to try next or what might be causing the issue.  After it failed, I have this dialog.

![Screen Shot 2020-12-11 at 11 19 05 AM](https://user-images.githubusercontent.com/75856364/101939815-c0c7ea80-3ba2-11eb-9600-ef991ce42de2.png)

**Comment by matt335672** [comment]
Hi @TDS1969 

On CentOS 8, the user is validated with PAM. The error you are seeing is simply saying that this process has failed. So you'll need to consult the authentication logs on the client to see what has appended.

It's been a long time since I was adminning this kind of system, but if I recall correctly, you have to configure SSSD to allow logins with RDP in some situations, particularly if you're using GPOs. More details on that are in `sssd-ad(5)`. If you crank up the SSSD logging you might get more info from the SSSD log file.

If you're running such a config, you might find other lesser-used services like `at` also won't work with your AD users.

Hope that's a bit useful.

**Comment by evanlinde** [comment]
On CentOS 8, I've had to add the line `allowed_users=anybody` to `/etc/X11/Xwrapper.config`. (I don't remember whether I had to create the file or not, but on my systems this line is the only line in the file.)

**Comment by matt335672** [comment]
@TDS1969 - have you got any more information on the problem as reported, or do you need any more help?

@evanlinde - thanks for the input. It's useful info, but I don't believe it's applicable here as the original report is with `Xvnc`, and we haven't even got as far as starting it yet.

**Comment by matt335672** [comment]
@evanlinde - your post got me thinking. 

You shouldn't have to create `/etc/X11/Xwrapper.config`. It's lowering the overall security of your system, as now anyone can run the X server.

You should be able to achieve the same result but more securely by editing `/etc/rddp/sesman.ini` and replacing:-

```
param=Xorg
```

with 

```
param=/usr/libexec/Xorg
```

I've just raised [bugzilla #1911807](https://bugzilla.redhat.com/show_bug.cgi?id=1911807) on the EPEL RPM, as it isn't to my mind an obvious thing to have to do.

**Comment by bsmojver** [comment]
As I mentioned in the RH bug, the reason wrapper config was necessary was the required patch of xrdp, because as it was (when I was first integrating all this into Fedora), xrdp would not even start with selinux on Fedora when using Xorg (i.e. the configuration suggested above did not work). I will revisit the whole thing when I get some time, just in case things changed. Xvnc requires none of this.

**Comment by evanlinde** [comment]
@matt335672 Thanks for pointing this out. I have applied the change you suggested in `/etc/xrdp/sesman.ini`, removed `/etc/X11/Xwrapper.config`, and restarted xrdp. Everything is working properly.

**Comment by bsmojver** [comment]
@evanlinde That was with SELinux in enforcing mode, correct? And all of the contexts your processes are running in are correct?

**Comment by evanlinde** [comment]
@bsmojver Yes, SELinux is set to Enforcing. I'm not sure about *correct* contexts -- the context I see for almost all processes related to my session is `system_u:system_r:unconfined_service_t:s0`; the exceptions are `systemd`, `(sd-pam)`, and one of the `dbus-daemon` processes.

**Comment by bsmojver** [comment]
Yeah, I just tested on RHEL7, RHEL8 and F33. I think this was all done initially because of RHEL7, which does not ship /usr/libexec/Xorg at all (from memory - the patches were done a very long time ago). So, a compromise was made to use Xorg everywhere. I'll split this up into different patches, where RHEL7 will have the old style stuff and everything else will use /usr/libexec/Xorg, with no need to have wrapper config.

**Comment by bsmojver** [comment]
Should be corrected in 0.9.12-2.el8, which has been submitted for testing.

**Comment by matt335672** [comment]
@bsmojver, @evanlinde - thanks very much for the proactive and quick turnaround on this. I don't doubt it will help EL8 and Fedora users going forwards.

I'll leave this PR open for a week or so in case @TDS1969 comes back to me regarding the original issue!
